### ISSUE TITLE
forwarding cpus directive for local executor to Docker

### ISSUE THREAD

**Issue by huguesfontenelle** [nextflow-io/nextflow#2497]
## New feature

I'm wondering, is there any reason why the `cpus` directive is not used with the local executor? (using the `--cpus` docker params) After all, the Docker param `--memory` is used.

## Usage scenario 

I'm using the local executor on a e2-standard-32 google VM (32 cpus, 128 GB ram). Sometimes, some processes seem to freeze or take too long when ran in parallel with other processes (even though they run fast on their own with their cpu/memory allocation). I've had this kind of issues with: hisat2, arriba, vardict.
Being a concurrency issue it is obviously quite hard to reproduce, but I can't help noticing that even though a process is allocated, say, 4 cpus in the config, there is actually no enforcement of this limit for Docker using the local executor.

## Suggest implementation 

A workaround may be
```
containerOptions "--cpus ${task.cpus}"
```
Haven't tried yet.

**Comment by abhi18av** [comment]
Hi @huguesfontenelle ,

If I understand the core request correctly, [the `--cpus` flag for the docker container](https://docs.docker.com/config/containers/resource_constraints/#configure-the-default-cfs-scheduler) does get populated in 

https://github.com/nextflow-io/nextflow/blob/909f95279277613c8bc9bef1481a0ef3ba38af65/modules/nextflow/src/test/groovy/nextflow/container/DockerBuilderTest.groovy#L137

and https://github.com/nextflow-io/nextflow/blob/909f95279277613c8bc9bef1481a0ef3ba38af65/modules/nextflow/src/main/groovy/nextflow/container/DockerBuilder.groovy#L120

Have you checked the resultant `.command.run` file within a task directory for a failing task?

**Comment by huguesfontenelle** [comment]
Thank you @abhi18av for answering so promptly and pointing me to the code.
Well yes I had checked `.command.run`, but that wasn't the issue. I failed to provide which version I was using (`20.10.0`)
Clearly, this feature is present in the latest version.
I believe that you introduced it in #1833 and shipped in `21.01.0-edge` :-)
I wasn't updating to the "edge" releases (but now I should).

Thank you. /closing (if I may)
